src/overlays/actors/ovl_En_Dnk/z_en_dnk.c

```diff
@@ -70,44 +133,336 @@ static DamageTable D_80A521FC = {
+    }
+
+    this->unk_2A2 = D_80A521A0;
+    D_80A521A0++;
+}
+
+void EnDnk_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnDnk* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnDnk_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnDnk* this = THIS;
+    s32 pad;
+
+    this->actionFunc(this, globalCtx);
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    func_80A515C4(this);
+    Actor_SetHeight(&this->actor, 34.0f);
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    func_80A52134(this, globalCtx);
+}
+
+s32 func_80A51A78(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    EnDnk* this = THIS;
+
+    this->unk_260[limbIndex] = *dList;
+    *dList = NULL;
+    return 0;
+}
+
+void func_80A51AA4(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    EnDnk* this = THIS;
+    MtxF sp5C;
+    Vec3f sp50 = D_801D15B0;
+    Vec3f sp44;
+    Vec3s sp3C;
+
+    if (limbIndex == 2) {
+        SysMatrix_MultiplyVector3fByState(&sp50, &sp44);
+        SysMatrix_CopyCurrentState(&sp5C);
+        func_8018219C(&sp5C, &sp3C, 0);
+        SysMatrix_InsertTranslation(sp44.x, sp44.y, sp44.z, MTXMODE_NEW);
+        Matrix_Scale(this->actor.scale.x, this->actor.scale.y, this->actor.scale.z, MTXMODE_APPLY);
+        if (this->unk_28C & 0x10) {
+            if (this->unk_28C & 0x20) {
+                sp3C.x = this->unk_296;
+                sp3C.y = this->unk_298;
+                sp3C.y += this->actor.shape.rot.y;
+                sp3C.z = 0;
+                Math_SmoothStepToS(&this->unk_290, sp3C.x, 4, 0x1FFE, 1);
+                Math_SmoothStepToS(&this->unk_292, sp3C.y, 4, 0x1FFE, 1);
+                Math_SmoothStepToS(&this->unk_294, sp3C.z, 4, 0x1FFE, 1);
+            } else {
+                this->unk_290 = sp3C.x;
+                this->unk_292 = sp3C.y;
+                this->unk_294 = sp3C.z;
+            }
+        } else {
+            this->unk_28C |= 0x10;
+            this->unk_290 = sp3C.x;
+            this->unk_292 = sp3C.y;
+            this->unk_294 = sp3C.z;
+        }
+
+        Matrix_RotateY(this->unk_292, MTXMODE_APPLY);
+        SysMatrix_InsertXRotation_s(this->unk_290, MTXMODE_APPLY);
+        SysMatrix_InsertZRotation_s(this->unk_294, MTXMODE_APPLY);
+    }
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(POLY_OPA_DISP++, this->unk_260[limbIndex]);
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
+
+void func_80A51CB8(EnDnk* this, GlobalContext* globalCtx) {
+    static Gfx* D_80A5245C[] = {
+        D_06001680,
+        D_06001700,
+        D_06001780,
+    };
```

Comment:
Are you sure those are DLists? 
Those are being set to segment 8, which is usally used to draw eyes or similar stuff (it's not always that tho).
Just by looking the code I can't tell if those are DLists or not. I suggest either checking the object or using `UNK_PTR`

Reply:
If it's eye textures the correct type would be `TexturePtr`